### PR TITLE
Support for Rust's native boolean types.

### PR DESCRIPTION
**Pull request by jacob-hughes**
Nice and straightforward PR. Necessary for upcoming MIR assertion evaluation.

### CODE REVIEW

File: src/interp.rs [hunk 1]
```diff
@@ -106,6 +115,14 @@ impl<'tcx> TyVal<'tcx> {
                 },
                 _ => unimplemented!()
             },
+            TypeVariants::TyBool => {
+                let mut val = false;
```
Comment (ltratt) [inline src/interp.rs]: This can just be `assert_eq!(...); let val = bytes[0] == 1`.

Comment (jacob-hughes) [reply src/interp.rs]: Fixed

**Comment by vext01** [comment]
LGTM. In the future it might be worth putting unrelated code formatting changes in a separate commit.

**Comment by jacob-hughes** [comment]
Okay, will try and do this in future. Am I okay to squash?

**Comment by vext01** [comment]
fine by me. @ltratt

**Comment by ltratt** [comment]
If you squash, I can merge.

**Comment by jacob-hughes** [comment]
Squashed!

**Pull request merged**